.github/workflows/run_all_frameworks.yml

```diff
@@ -0,0 +1,128 @@
+          echo Common files: $common_files
+          echo Changed files: $changed_files
+
+          is_common=1
+          for f in $changed_files
+          do
+            if echo $common_files | grep -e $f >> /dev/null;
+            then
+                echo "File $f detected as common file"
+                is_common=0
+                break
+            fi
+          done
+
+          # Indicates which jobs should be executed (0) or not (1)
+          skip_evaluation=0
+          if [[ $is_common -eq 0 ]];
+          then
+            FRAMEWORKS='["autogluon", "autosklearn", "gama", "h2oautoml", "mlplanweka", "tpot", "constantpredictor", "randomforest", "tunedrandomforest"]'
+          else
+            changed_frameworks=$(git diff --name-only HEAD..$GITHUB_BASE_REF | grep -o -i -P 'frameworks/(?!shared).*/' | uniq | sed -e 's/frameworks//' -e 's/\///g')
+            if [ ! -z $changed_frameworks ];
+            then
+              json_array=[
+              for framework in $changed_frameworks; do json_array=$json_array\"$framework\",; done
+              FRAMEWORKS=${json_array::-1}]  #remove trailing comma and add closing bracket
+            else
+              # No changes to common files or frameworks - must be e.g. docs. No need to run tests.
+              skip_evaluation=1
+              FRAMEWORKS=[]
+            fi
+          fi
+
+          echo Building matrix for frameworks: $FRAMEWORKS
+
+          echo "::set-output name=frameworks::$FRAMEWORKS"
+          echo "::set-output name=skip_baseline::$is_common"
+          echo "::set-output name=skip_evaluations::$skip_evaluation"
+
+  baseline:
+    name: Run Baseline on OpenML Iris
+    runs-on: ubuntu-latest
+    needs: detect_changes
+    if: needs.detect_changes.outputs.skip_baseline == 0
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            pip-
+      - name: Setup Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - name: Install Requirements
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+      - name: Run constantpredictor on openml iris
+        run: |
+          python runbenchmark.py constantpredictor openml/t/59 1h4c
+      - name: Echo Python Files
+        run: |
+          python -c "import numpy; print(numpy.__file__)"
+
+  run_frameworks:
+    name: ${{ matrix.framework }}/${{ matrix.task }}
+    runs-on: ubuntu-latest
+    needs:
+     - baseline
+     - detect_changes
+    if: ${{ success() }} || ${{ cancelled() }}
+    strategy:
+      matrix:
+        python-version: [3.8]
+        framework: ${{ fromJson(needs.detect_changes.outputs.frameworks) }}
+        task: [APSFailure, bioresponse, dresses-sales, eucalyptus, internet-advertisements, kc1, micro-mass]
```

Comment:
This should probably be changed to fetch from `validation.yaml`, but developing so far has been a bit of a PITA so I hope it's okay to keep it hard-coded for now.

Reply:
I'm fine with hardcoded, it's not as if we're going to change `validation.yaml` often.

Actually wondering if we always need all of them: I would run them all for framework-specific changes, and a subset (2 or 3 max) for common changes given that we have a much larger test matrix with all frameworks being included: I would even stick to `test` datasets in this case (shorter time constraints + one dataset of each kind).
wdyt?

Reply:
I think that's a good idea (test for shared, validation for specific). Shared changes which affect a validation dataset but none of the test sets are probably unlikely enough to justify the additional compute time. Will update.